src/khoj/routers/api.py

```diff
@@ -480,17 +497,17 @@ def _save_to_conversation_log(
             status_code=500, detail="Set your OpenAI API key via Khoj settings and restart it to use Khoj Chat."
         )
 
+
+async def _get_top_k_results(
```

Comment:
Maybe we can just extract search queries in this function and call it `extract_search_queries` or some such for better modularity (and naming). The code snippets that calls the `search` function can then just be in the main `chat` function?